cpp/src/arrow/array/data.cc

```diff
@@ -236,6 +236,18 @@ void SetOffsetsForScalar(ArraySpan* span, offset_type* buffer, int64_t value_siz
   span->buffers[buffer_index].size = 2 * sizeof(offset_type);
 }
 
+template <typename RunEndType>
+void FillRunEndsArrayForScalar(ArraySpan* span, const DataType* run_end_type) {
```

Comment:
Are these changes related for this PR?

Reply:
If so, can you perhaps add a test for `FillFromScalar` on a run-end-encoded scalar? For example in `arrow/array/array_run_end_test.cc`.

Reply:
Yes, the tests for scalar kernels automatically run the kernel also on actual scalars, and then this ends up creating an array from the scalar. So those changes to `FillFromScalar` are needed to be able to run any kernel on a REE scalar.

At the moment we don't actually have any test for `FillFromScalar` directly (also not for other types), only through testing the kernels on scalars.